### PR TITLE
feat: Give TimeShift an update button

### PR DESCRIPTION
**Pull request by eseidel**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+import 'package:time_shift/update_button.dart';
```
Comment (felangel) [inline lib/main.dart:3]: Did we forget to commit this file?

**Comment by eseidel** [comment]
Fixed, sorry.

File: lib/main.dart (hunk 1 above)
Comment (eseidel) [reply lib/main.dart:3]: Fixed, sorry.

**Review by bryanoltman — APPROVED** [review]

File: lib/update_button.dart [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  UpdateStatus status = UpdateStatus.idle;
+  late bool _haveShorebirdEngine;
+  bool _haveUpdate = false;
+  final _shorebirdCodePush = ShorebirdCodePush();
+
+  @override
+  void initState() {
+    super.initState();
+    final ffi.DynamicLibrary library = ffi.DynamicLibrary.process();
+    _haveShorebirdEngine = library.providesSymbol('shorebird_update');
```
Comment (felangel) [inline lib/update_button.dart:32]: Minor but would be nice to expose this from the code push package

**Review by felangel — APPROVED** [review]

File: lib/update_button.dart (hunk 2 above)
Comment (eseidel) [reply lib/update_button.dart:32]: 💯  https://github.com/shorebirdtech/updater/issues/49

**Pull request merged**